### PR TITLE
Fix watcher invocation with WatchIgnorePlugin

### PR DESCRIPTION
**Pull request by lemonmade**
As noted in https://github.com/webpack/webpack/pull/9097 and https://github.com/sysgears/webpack-virtual-modules/issues/15, the way `WatchIgnorePlugin` works breaks some of the internal API usage this package uses. Both issues have indicated it's essentially a "wontfix", pointing at the other as the location for the fix.

I'm fine if this gets declined (I'll have to continue using a fork, though), but my perspective is that, since this package is using private API within Webpack, it should be willing to navigate all private API, including some that is only conditionally present. This encompasses the `WatchIgnorePlugin`, which is a first-party plugin provided by Webpack, and so in this PR I have checked for, and conditionally used, the private API of that plugin to get access to the underlying watch file system.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,9 +46,18 @@ VirtualModulesPlugin.prototype.writeModule = function(filePath, contents) {
 
   debug(self._compiler.name, "Write module:", modulePath, contents);
 
+  // When using the WatchIgnorePlugin (https://github.com/webpack/webpack/blob/52184b897f40c75560b3630e43ca642fcac7e2cf/lib/WatchIgnorePlugin.js),
+  // the original watchFileSystem is stored in `wfs`. The following "unwraps" the ignoring
+  // wrappers, giving us access to the "real" watchFileSystem.
+  let finalWatchFileSystem = self._watcher && self._watcher.watchFileSystem;
+
+  while (finalWatchFileSystem && finalWatchFileSystem.wfs) {
```
Comment (vlasenko) [inline index.js]: Thanks, the general approach looks good to me. I'm wonder however, why do we need while loop here?

Comment (lemonmade) [reply index.js]: When you use multiple `WatchIgnorePlugin`, each of them overwrites the `watchFileSystem`, using the previous `watchFileSystem` as `wfs`: https://github.com/webpack/webpack/blob/52184b897f40c75560b3630e43ca642fcac7e2cf/lib/WatchIgnorePlugin.js#L98-L101

Comment (vlasenko) [reply index.js]: Aha, I see, thank you for clarification

**Comment by vlasenko** [comment]
@lemonmade Merged and published as `webpack-virtual-modules@0.1.12`. Many thanks for your effort!

**Comment by lemonmade** [comment]
Thanks so much for the quick review and publish!

**Comment by vlasenko** [comment]
Thank you for finding a way to resolve this issue and providing a PR!

**Comment by ChrisSG** [comment]
Thx lemonmade!

**Pull request merged**